lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,461 @@
+        var buffer = self
+        while buffer > 0 {
+            var byte = UInt8(buffer & 0b0111_1111)
+            buffer >>= 7
+            if buffer > 0 {
+                byte |= 0b1000_0000
+            }
+            bytes.append(byte)
+        }
+        return bytes
+    }
+}
+
+// MARK: - Diagnostics
+
+fileprivate struct ErrorMessage: DiagnosticMessage, ExpressibleByStringInterpolation {
+    init(stringLiteral value: String) {
+        self.message = value
+    }
+    var message: String
+    var diagnosticID: MessageID { .init(domain: "DebugDescription", id: "DebugDescription")}
+    var severity: DiagnosticSeverity { .error }
+}
+
+extension MacroExpansionContext {
+    fileprivate func diagnose(node: some SyntaxProtocol, error message: ErrorMessage) {
+        diagnose(Diagnostic(node: node, message: message))
+    }
+}
+
+// MARK: - Syntax Tree Helpers
+
+extension MacroExpansionContext {
+    /// Determine the module name of the Syntax node, via its fileID.
+    /// See https://developer.apple.com/documentation/swift/fileid()
+    fileprivate func moduleName<T: SyntaxProtocol>(of node: T) -> String? {
+        if let fileID = self.location(of: node)?.file.as(StringLiteralExprSyntax.self)?.representedLiteralValue,
+           let firstSlash = fileID.firstIndex(of: "/") {
+            return String(fileID.prefix(upTo: firstSlash))
+        }
+        return nil
+    }
+}
+
+extension DeclGroupSyntax {
+    fileprivate var typeName: String? {
+        switch self.kind {
+        case .actorDecl, .classDecl, .enumDecl, .structDecl:
+            return self.asProtocol(NamedDeclSyntax.self)?.name.text
+        case .extensionDecl:
+            return self.as(ExtensionDeclSyntax.self)?.extendedType.description
+        default:
+            break
+        }
+        return nil
+    }
+}
+
+extension SyntaxKind {
+    fileprivate var declName: String {
+        var name = String(describing: self)
+        name.removeSuffix("Decl")
+        return name
+    }
+}
+
+extension String {
+    fileprivate mutating func removeSuffix(_ suffix: String) {
+        if self.hasSuffix(suffix) {
+            return self.removeLast(suffix.count)
+        }
+    }
+}
+
+extension PatternBindingSyntax {
+    /// The property's name.
+    fileprivate var name: String? {
+        self.pattern.as(IdentifierPatternSyntax.self)?.identifier.text
+    }
+
+    /// Predicate which identifies computed properties.
+    fileprivate var isComputedProperty: Bool {
+        guard let accessors = self.accessorBlock?.accessors else {
+            // No accessor block, not computed.
+            return false
+        }
+
+        switch accessors {
+        case .accessors(let accessors):
+            // A `get` accessor indicates a computed property.
+            return accessors.contains { $0.accessorSpecifier.tokenKind == .keyword(.get) }
+        case .getter:
+            return true
+        @unknown default:
+            return true
+        }
```

Comment:
FWIW you could join the `nil` case into the `switch`

```suggestion
        switch self.accessorBlock?.accessors {
        case .accessors(let accessors)?:
            // A `get` accessor indicates a computed property.
            return accessors.contains { $0.accessorSpecifier.tokenKind == .keyword(.get) }
        case .getter?:
            return true
        case nil:
            // No accessor block, not computed.
            return false
        @unknown default:
            return true
        }
```